lib/css-converter.js

```diff
@@ -0,0 +1,257 @@
+import { CssSelectorParser } from 'css-selector-parser';
+import log from './logger';
+
+const CssConverter = {};
+
+const parser = new CssSelectorParser();
+parser.registerSelectorPseudos('has');
+parser.registerNestingOperators('>', '+', '~');
+parser.registerAttrEqualityMods('^', '$', '*', '~');
+parser.enableSubstitutes();
+
+const booleanAttrs = [
+  'checkable', 'checked', 'clickable', 'enabled', 'focusable',
+  'focused', 'long-clickable', 'scrollable', 'selected',
+];
+
+const numericAttrs = [
+  'index', 'instance',
+];
+
+const strAttrs = [
+  'description', 'resource-id', 'text', 'class-name', 'package-name'
+];
+
+const allAttrs = [
+  ...booleanAttrs,
+  ...numericAttrs,
+  ...strAttrs,
+];
+
+const attributeAliases = [
+  ['resource-id', ['id']],
+  ['description', [
+    'content-description', 'content-desc',
+    'desc', 'accessibility-id',
+  ]],
+  ['index', ['nth-child']],
+];
+
+/**
+ * Convert hyphen separated word to snake case
+ * @param {String} str
+ */
+function toSnakeCase (str) {
+  const tokens = str.split('-').map((str) => str.charAt(0).toUpperCase() + str.slice(1).toLowerCase());
+  let out = tokens.join('');
+  return out.charAt(0).toLowerCase() + out.slice(1);
+}
+
+/**
+ * Get the boolean from a CSS object. If empty, return true. If not true/false/empty, throw exception
+ *
+ * @param {*} css CSS object
+ */
+function assertGetBool (css) {
+  const val = css.value?.toLowerCase() || 'true';
```

Comment:
why `|| 'true'` ?

Reply:
@mykola-mokhnach good question...

Answer with an example:
* In css `inputs[checked]` implies that you're matching inputs that are checked (ie: where `checked=true`. So if the value is omited, it's implied that `checked=true`

Reply:
* `inputs[checked=true]` -> `checked(true)`
* `inputs[checked=false]` -> `checked(false)`
* `inputs[checked]` -> `checked(true)`

Reply:
Please add the explanation above as a comment